.travis.yml

```diff
@@ -17,47 +6,92 @@ env:
-- stack test --ta -j1
-- if [ "$TRAVIS_BRANCH" == "master" ]; then source deploy/deploy.sh; fi
+stages:
+  - "Build"
+  - name: "Deploy"
+    if: branch = master
+    
+jobs:
+  include:
+    - name: "Web"
+      stage: "Build"
+      language: node_js
+      node_js:
+        - "11"
+      cache: npm
+      services:
+        - docker
+      env:
+        DOCKER_TAG: ${DOCKER_REPO}/mirza-web:$TRAVIS_COMMIT
+      before_install:
+        - cd ./web
+      install:
+        - npm install
+      script:
+        - ./build.sh
+        - |
+          if [ "$TRAVIS_BRANCH" == "master" ]; then
+            docker build -t "${DOCKER_TAG}" .
+            ../deploy/docker_push.sh "${DOCKER_TAG}"
+          fi
+    
+    - name: "Back-end"
+      stage: "Build"
+      language: generic
+
+      services:
+        - postgresql
+        - docker
+
+      # Caching so the next build will be fast too.
+      cache:
+        timeout: 4000
+        directories:
+          - "$HOME/.stack"
+          - ".stack-work"
+
+      # Ensure necessary system libraries are present
+      addons:
+        apt:
+          packages:
+            - libgmp-dev
+      env:
+        - SCS_DOCKER_TAG: ${DOCKER_REPO}/supplychainserver:$TRAVIS_COMMIT
+        - BR_DOCKER_TAG: ${DOCKER_REPO}/businessregistry:$TRAVIS_COMMIT
+      before_install:
+        # Download and unpack the stack executable
+        - mkdir -p ~/.local/bin
+        - export PATH=$HOME/.local/bin:$PATH
+        - wget -qO- https://get.haskellstack.org/ | sh
+
+      install:
+        # Build dependencies
+        - stack --no-terminal --install-ghc test --only-dependencies
+
+      script:
+        # Build the package, its tests, and its docs and run the tests
+        - stack clean
+        - stack --no-terminal build --copy-bins
+        - stack ls dependencies --license | ./license_check.py
+        - psql -c 'CREATE DATABASE testsupplychainserver;' -U postgres
+        - psql -c 'CREATE DATABASE testbusinessregistry;' -U postgres
+        - stack exec supplyChainServer -- --init-db -c 'dbname=testsupplychainserver'
+        - echo 'YES' | stack exec businessRegistry -- initdb -c 'dbname=testbusinessregistry'
+        - stack test --ta -j1
+        - |
+          if [ "$TRAVIS_BRANCH" == "master" ]; then
+            docker build -f Mirza.Dockerfile --target PKG-SCS -t "${SCS_DOCKER_TAG}" .
+            docker build -f Mirza.Dockerfile --target PKG-BR -t "${BR_DOCKER_TAG}" .
+            ./deploy/docker_push.sh "${SCS_DOCKER_TAG}"
+            ./deploy/docker_push.sh "${BR_DOCKER_TAG}"
+          fi
+
+    - name: "Deploy to AWS"
+      stage: "Deploy"
+      before_install:
+        # Decrypt the ssh key that will be used for deployments 
+        - openssl aes-256-cbc -K $encrypted_8fe1f88deb69_key -iv $encrypted_8fe1f88deb69_iv
+          -in deploy/deployment.pem.enc -out ~/.ssh/deployment.pem -d
+          
+      script:
```

Comment:
This might also need `if [[ "${TRAVIS_PULL_REQUEST}" = "false" ]]`. Not 100% sure though.

According to the travis if you specify it as a deploy stage it should be skipped in PR's automatically, but we are defining this stage normally instead of as a deploy stage.

See docco here: https://docs.travis-ci.com/user/deployment#pull-requests